src/Workspaces/CoreTest/WorkspaceTests/WorkspaceTests.cs

```diff
@@ -224,6 +169,26 @@ public Document AddDocument(DocumentInfo documentInfo)
             }
         }
 
+        [Fact]
+        public void ImmutableEmptyWorkspace_TryApplyChanges()
```

Comment:
Is the point of this test just to make sure somebody doesn't mess with it somehow by accident? It might be worth a comment saying this test is really only testing that shim, and if the shim goes away this could also just be deleted.